src/classes/FileManager.java

```diff
@@ -0,0 +1,126 @@
+package classes;
+
+import java.io.*;
+import java.util.List;
+
+/**
+ * Created by aloginov on 14.09.16.
+ */
+public class FileManager {
+
+    private File file;
+    private BufferedReader in;
+    private BufferedWriter out;
```

Comment:
In your case, it seems, you spend a lot of memory trying to put large file into the cache.
See http://www.baeldung.com/java-read-lines-large-file